src/Hedgehog/Property.fs

```diff
@@ -418,15 +418,17 @@ module Property =
     let recheckBool (size : Size) (seed : Seed) (g : Property<bool>) : unit =
         bind g ofBool |> recheck size seed
 
+    let toString' (n : int<tests>) (p : Property<unit>) : string =
+        Report.render <| report' n p
+
+    let toString (p : Property<unit>) : string =
+        Report.render <| report p
+
     let print' (n : int<tests>) (p : Property<unit>) : unit =
-        report' n p
-        |> Report.render
-        |> printfn "%s"
+        printfn "%s" <| toString' n p
```

Comment:
Just a thought... Perhaps `asString` instead of `toString`? (/cc @TysonMN) — In the Hedgehog.Linq namespace, for example, `ToString` may cause confusion because of the built-in [ToString](https://docs.microsoft.com/en-us/dotnet/api/system.object.tostring) that all types inherit.

Reply:
What is it called in the Haskell code? `render`?

Reply:
Overriding `ToString` is actually recommended in C#. From [the docs](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/how-to-override-the-tostring-method):

>When you create a custom class or struct, you should override the ToString method in order to provide information about your type to client code.

I think `toString`/`ToString` is fine, and since we're in dotnet land, I think we should work within their conventions (and not Haskell's - at least in this matter).

Reply:
I mostly indifferent about this naming.  What @dharmaturtle says sounds good to me.

Reply:
Maybe we should just override `ToString` for these types, and not make it part of the LINQ interface?

Reply:
Then I no longer want to call this `ToString`.  I want to prioritise #F conventions (like separating types from behavior) over C# ones (like combing types and behavior).

Reply:
Renamed all instances to `asString`/`AsString`

Reply:
[Someone else ran into this problem.](https://github.com/fsharp/fslang-suggestions/issues/691)

I'm fine with `asString`.

Reply:
> What is it called in the Haskell code? `render`?

IIRC, it's called `sample` the equivalent of the one we're tackling here.